src/metatrain/utils/additive/_base_composition.py

```diff
@@ -0,0 +1,552 @@
+"""
+Contains the ``BaseCompositionModel class. This is intended for eventual porting to
+metatomic. The class ``MetatrainCompositionModel`` wraps this to be compatible with
+metatrain-style objects.
+"""
+
+from typing import Dict, List, Optional
+
+import metatensor.torch as mts
+import torch
+from metatensor.torch import Labels, LabelsEntry, TensorBlock, TensorMap
+from metatensor.torch.learn.data import DataLoader
+from metatomic.torch import ModelOutput, System
+
+
+class BaseCompositionModel(torch.nn.Module):
+    """
+    Fits a composition model for a dict of targets.
+
+    A composition model is a model that predicts the target values based on the
+    composition of the system, i.e., the chemical identity of atoms in the system.
+
+    Only invariant blocks of the specified targets are fitted, i.e. those indexed by
+    keys with a single name "_" (for scalars) or keys with including names "o3_lambda"
+    and "o3_sigma" (for spherical targets).
```

Comment:
Are you only fitting a model for `o3_lambda=0, o3_sigma=1`? If so, could you please make this explicit here?